src/Compilers/Core/Portable/InternalUtilities/PerformanceSensitiveAttribute.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Roslyn.Utilities
+{
+    /// <summary>
+    /// Indicates that a code element is performance sensitive under a known scenario.
+    /// </summary>
+    /// <remarks>
+    /// <para>When applying this attribute, only explicitly set the values for properties specifically indicated by the
+    /// test/measurement technique described in the associated <see cref="Uri"/>.</para>
+    /// </remarks>
+    [Conditional("EMIT_CODE_ANALYSIS_ATTRIBUTES")]
+    [AttributeUsage(AttributeTargets.Constructor | AttributeTargets.Method | AttributeTargets.Property | AttributeTargets.Field, AllowMultiple = true, Inherited = false)]
+    internal sealed class PerformanceSensitiveAttribute : Attribute
```

Comment:
This attribute should also be applied to this locations to avoid regressions:
PR #23490
[SourceNamespaceSymbol](https://github.com/sharwell/roslyn/blob/63ffc585a6f0234564075d6cdc9d4ebd6295ba91/src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs#L466) 
[IsDefinedInSourceTree](https://github.com/sharwell/roslyn/blob/63ffc585a6f0234564075d6cdc9d4ebd6295ba91/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs#L873)

Reply:
➡️ Filed #23615